cmd/lifecycle/cli/command.go

```diff
@@ -47,22 +48,23 @@ func Run(c Command, withPhaseName string, asSubcommand bool) {
 			cmd.Exit(err)
 		}
 	}
-	cmd.DisableColor(noColor)
 
 	if printVersion {
 		cmd.ExitWithVersion()
 	}
-	if err := cmd.DefaultLogger.SetLevel(logLevel); err != nil {
-		cmd.Exit(err)
-	}
-	cmd.DefaultLogger.Debugf("Starting %s...", withPhaseName)
 
 	for _, arg := range flagSet.Args() {
 		if arg[0:1] == "-" {
 			cmd.DefaultLogger.Warnf("Warning: unconsumed flag-like positional arg: \n\t%s\n\t This will not be interpreted as a flag.\n\t Did you mean to put this before the first positional argument?", arg)
 		}
 	}
 
+	cmd.DisableColor(c.Inputs().NoColor)
+	if err := cmd.DefaultLogger.SetLevel(c.Inputs().LogLevel); err != nil {
+		cmd.Exit(err)
+	}
+	cmd.DefaultLogger.Debugf("Starting %s...", withPhaseName)
```

Comment:
Hmm I don't know why I moved this - we should disable color before we print the warning